marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/job/startStep.sjs

```diff
@@ -17,9 +17,55 @@
 
 xdmp.securityAssert("http://marklogic.com/data-hub/privileges/run-step", "execute");
 
+const config = require("/com.marklogic.hub/config.sjs");
+const coreLib = require('/data-hub/5/artifacts/core.sjs');
+const dhProv = require('/data-hub/5/provenance/dh-provenance.xqy');
+const hubUtils = require("/data-hub/5/impl/hub-utils.sjs");
 const Job = require("/data-hub/5/flow/job.sjs");
+const jobs = require("/data-hub/5/impl/jobs.sjs");
 
 var jobId;
 var stepNumber;
+var flowName;
 
-Job.getRequiredJob(jobId).startStep(stepNumber).update()
\ No newline at end of file
+function updateProvenance(flowName, stepNumber, jobId) {
+  const stepDetails = fn.head(hubUtils.invokeFunction(function () {
+    const fullFlow = coreLib.getFullFlow(flowName);
+    return fullFlow["steps"][stepNumber];
+  }, config.STAGINGDATABASE));
+
+  const latestProvenance = stepDetails["options"]["latestProvenance"];
```

Comment:
This is ignoring runtime options, but I'm not sure if that matters for this release. The use of getFullFlow is also a little inefficient, because that's doing a query for all steps, while we only need one. getFullFlow (which should really be in flow.sjs, not core.sjs, but that's a separate matter) could accept an optional stepNumber argument to only query for that step. 

But the main issue is runtime options - but only if we're exposing this property for 5.6. If we're not, I think it would be worth adding a 5.7 ticket so we don't forget to support runtime options here.

Reply:
@rahulvudutala How difficult would it be for you to adjust this to use the runtime options as part of this PR? Beyond needing to support it eventually, it would be more convenient for demos and testing if it was incorporated in this PR.

Reply:
@rjrudin @ryanjdew Added changes taking care of runtime options and getFullFlow only for a step